ls1intum/Artemis
# [Bugfix] Trim title of an exercise before saving it

### Checklist

### Motivation and Context
(Accidentally) adding whitespace at the end of an exercise title might lead to confusion. When the user wants to delete an exercise and is prompted to enter its title, he won't be able to confirm the delete operation without entering the extra whitespace at the end.

This bug has been reported in #2035.

### Description
Before sending a request to the backend when saving an exercise, we trim the exercise title in the frontend. Additionally, we also make sure to only persist exercises with trimmed titles in the backend.

### Steps for Testing
1. Create a (modeling) exercise with whitespaces at the end of the title (e.g. 'My Exercise ')
2. Delete the exercise
3. You should be able to confirm the delete operation by entering the trimmed title (e.g. 'My Course')

### Screenshot
<img width="735" alt="Screenshot 2020-08-31 at 16 48 31" src="https://user-images.githubusercontent.com/21973908/91733565-f9baf100-eba9-11ea-8313-c0bc0137a5c1.png">
